components/tidb_query_executors/src/top_n_executor.rs

```diff
@@ -309,6 +338,12 @@ impl<Src: BatchExecutor> BatchExecutor for BatchTopNExecutor<Src> {
             };
         }
 
+        if let Some(paging_size) = self.context.cfg.paging_size {
+            if self.n > paging_size as usize {
```

Comment:
if `n > paging_size > batch_size`, only one batch will be scanned. The cop loop then scans the executor again because `record_all < paging_size` calls the topn executor and reads more data and concatenates them together. I'm worried if the total order can be guaranteed in the concatenated chunk.